browser/typescript/v2/react-app/src/ampli/index.ts

```diff
@@ -0,0 +1,747 @@
+      return false;
+    }
+    return !this.disabled;
+  }
+
+  /**
+   * Initialize the Ampli SDK. Call once when your application starts.
+   * @param options Configuration options to initialize the Ampli SDK with.
+   */
+  load(options?: LoadOptions) {
+    this.disabled = options?.disabled ?? false;
+
+    if (this.amplitude) {
+      console.warn('WARNING: Ampli is already intialized. Ampli.load() should be called once at application startup.');
+      return;
+    }
+
+    const env = options?.environment ?? 'development';
+    const apiKey = options?.client?.apiKey ?? ApiKey[env];
+
+    if (options?.client?.instance) {
+      this.amplitude = options?.client?.instance;
+    } else if (apiKey) {
+      this.amplitude = amplitude;
+      return this.amplitude?.init(apiKey, undefined, { ...DefaultOptions, ...options?.client?.options });
+    } else {
+      console.error("ERROR: ampli.load() requires 'environment', 'client.apiKey', or 'client.instance'");
+    }
+  }
+
+  /**
+   * Identify a user and set user properties.
+   *
+   * @param userId The user's id.
+   * @param properties The user properties.
+   * @param options Optional event options.
+   */
+  identify(
+    userId: string | undefined,
+    properties: IdentifyProperties,
+    options?: EventOptions,
+  ) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    if (userId) {
+      options = Object.assign({}, options, { user_id: userId });
+    }
+
+    const amplitudeIdentify = new amplitude.Identify();
+    if (properties != null) {
+      for (const [key, value] of Object.entries(properties)) {
+        amplitudeIdentify.set(key, value);
+      }
+    }
+    return this.amplitude?.identify(
+      amplitudeIdentify,
+      options,
+    );
+  }
+
+  setGroup(name: string, value: string | string[]) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    this.amplitude?.setGroup(name, value);
+  }
+
+  /**
+   * Identify a group and set group properties.
+   *
+   * @param groupType The group type.
+   * @param groupName The group name.
+   * @param properties The group properties.
+   * @param options Optional event options.
+   */
+  groupIdentify(
+    groupType: string,
+    groupName: string | string[],
+    properties: GroupProperties,
+    options?: EventOptions,
+  ) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    const amplitudeIdentify = new amplitude.Identify();
+    if (properties != null) {
+      for (const [key, value] of Object.entries(properties)) {
+        amplitudeIdentify.set(key, value);
+      }
+    }
+    return this.amplitude?.groupIdentify(groupType, groupName, amplitudeIdentify, options);
+  }
+
+  /**
+   * Track event
+   *
+   * @param event The event to track.
+   * @param options Optional event options.
+   */
+  track(event: Event, options?: EventOptions) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    return this.amplitude?.track(
```

Comment:
is this interface correct?

Reply:
It is correct, but I will remove `event.event_properties` because it is not needed.